catalog/app/containers/Admin/Settings/ThemeEditor.tsx

```diff
@@ -0,0 +1,449 @@
+                      errors={{
+                        url: 'Image should be valid url',
+                      }}
+                      disabled={submitting}
+                      fullWidth
+                      InputLabelProps={{ shrink: true }}
+                    />
+                  ) : (
+                    <RF.Field
+                      component={InputFile}
+                      initialValue={settings?.logo?.url || ''}
+                      name="logoUrl"
+                      label="Logo URL"
+                      placeholder="e.g. https://example.com/path.jpg"
+                      validate={
+                        validators.composeOr(
+                          validators.file,
+                          validators.url,
+                        ) as FF.FieldValidator<string>
+                      }
+                      errors={{
+                        url: 'Image should be valid url',
+                        file: 'Image should be file',
+                      }}
+                      disabled={submitting}
+                      fullWidth
+                      InputLabelProps={{ shrink: true }}
+                    />
+                  )}
+                  <M.Box pt={2} />
+                  <RF.Field
+                    // @ts-expect-error
+                    component={InputColor}
+                    initialValue={settings?.theme?.palette?.primary?.main || ''}
+                    name="primaryColor"
+                    label="Background color"
+                    placeholder="#282b50"
+                    validate={validators.hexColor as FF.FieldValidator<string>}
+                    errors={{
+                      hex: 'Enter 6-digit hex color, ex. #282b50',
+                    }}
+                    disabled={submitting}
+                    fullWidth
+                    InputLabelProps={{ shrink: true }}
+                  />
+                  <input type="submit" style={{ display: 'none' }} />
+                </form>
+              </M.DialogContent>
+              <M.DialogActions>
+                {submitting ? (
+                  <SubmitSpinner />
+                ) : (
+                  (!!error || !!submitError) && (
+                    <M.Box flexGrow={1} display="flex" alignItems="center" pl={2}>
+                      <M.Icon color="error">error_outline</M.Icon>
+                      <M.Box pl={1} />
+                      <M.Typography variant="body2" color="error">
+                        {error || submitError}
+                      </M.Typography>
+                    </M.Box>
+                  )
+                )}
+
+                <M.Button
+                  type="submit"
+                  onClick={handleSubmit}
+                  variant="contained"
+                  color="primary"
+                  disabled={submitting || (submitFailed && hasValidationErrors)}
+                >
+                  Save
```

Comment:
as per convention we use (based on mui guidelines, probably), "save" should be the rightmost button